src/meta/src/rpc/server.rs

```diff
@@ -171,315 +153,244 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
-    let notification_srv = NotificationServiceImpl::new(
-        env.clone(),
-        catalog_manager,
-        cluster_manager.clone(),
-        hummock_manager.clone(),
-        fragment_manager.clone(),
-        backup_manager.clone(),
-    );
-    let health_srv = HealthServiceImpl::new();
-    let backup_srv = BackupServiceImpl::new(backup_manager);
-
-    if let Some(prometheus_addr) = address_info.prometheus_addr {
-        MetricsManager::boot_metrics_service(
-            prometheus_addr.to_string(),
-            meta_metrics.registry().clone(),
-        )
+                .expect(err_msg),
+            );
+        } // sleep duration of election cycle, not fixed duration
+        sleep(Duration::from_secs(6)).await;
+        node_controllers
     }
 
-    let compaction_scheduler = Arc::new(CompactionScheduler::new(
-        env.clone(),
-        hummock_manager.clone(),
-        compactor_manager.clone(),
-    ));
-
-    // sub_tasks executed concurrently. Can be shutdown via shutdown_all
-    let mut sub_tasks =
-        hummock::start_hummock_workers(vacuum_manager, compaction_scheduler, &env.opts);
-    sub_tasks.push(
-        ClusterManager::start_worker_num_monitor(
-            cluster_manager.clone(),
-            Duration::from_secs(env.opts.node_num_monitor_interval_sec),
-            meta_metrics.clone(),
-        )
-        .await,
-    );
-    sub_tasks.push(HummockManager::start_compaction_heartbeat(hummock_manager).await);
-    sub_tasks.push((election_handle, election_shutdown));
-    if cfg!(not(test)) {
-        sub_tasks.push(
-            ClusterManager::start_heartbeat_checker(cluster_manager, Duration::from_secs(1)).await,
-        );
-        sub_tasks.push(GlobalBarrierManager::start(barrier_manager).await);
-    }
-    let (idle_send, idle_recv) = tokio::sync::oneshot::channel();
-    sub_tasks.push(
-        IdleManager::start_idle_checker(env.idle_manager_ref(), Duration::from_secs(30), idle_send)
-            .await,
-    );
-    let shutdown_all = async move {
-        for (join_handle, shutdown_sender) in sub_tasks {
-            if let Err(_err) = shutdown_sender.send(()) {
-                // Maybe it is already shut down
-                continue;
-            }
-            if let Err(err) = join_handle.await {
-                tracing::warn!("Failed to join shutdown: {:?}", err);
+    #[tokio::test]
+    async fn test_single_leader_setup() {
+        // make n also pseudo random?
+        let n = 5;
+        let _node_controllers = setup_n_nodes(n).await;
+
+        let mut leader_count = 0;
+        for i in 0..n {
+            // create client connecting against meta_i
+            let port = 1234 + i;
+            let meta_addr = format!("http://127.0.0.1:{}", port);
+            let host_addr = "127.0.0.1:5688".parse::<HostAddr>().unwrap();
+            let host_addr = HostAddr {
+                host: host_addr.host,
+                port: host_addr.port + i, // Do I need to change the port?
```

Comment:
```suggestion
                host: "127.0.0.1",
                port: 1234 + i,
```

Reply:
I am doing something similar in the next iteration of this PR by passing variables around. Let me know if that approach LGTY.